decidim-initiatives/app/services/decidim/initiatives/initiative_search.rb

```diff
@@ -65,7 +64,7 @@ def search_type_id
 
       def search_author
         if author == "myself" && options[:current_user]
-          query.where(decidim_author_id: options[:current_user].id)
+          Decidim::Initiative.where(decidim_author_id: options[:current_user], decidim_author_type: Decidim::UserBaseEntity.name)
```

Comment:
Shouldn't we keep using the existing `query` to combine with other filters?
```suggestion
          query.where(decidim_author_id: options[:current_user], decidim_author_type: Decidim::UserBaseEntity.name)
```

Reply:
I didn't reuse `query` because is fetching only published initiatives, and we have to show any initiative (published or not) created by the author when using the `my initiatives` filter 😃

Reply:
but have you tried to combine this filter with other filters?
At the moment Searchlight executes this method, constraints added by previously executed `search_*` methods will be lost. Check https://github.com/nathanl/searchlight/blob/d0a48fc3f1dd67538d7d412bf685ba46186374c8/lib/searchlight/search.rb#L66
the executed query is the one returned by the last `search*` method.
@marinavega

Reply:
Morning! Sorry I haven't seen this before. Not sure if I understand what you're suggesting: you mean adding filters on top of `query`? 🤔

Reply:
It seems that searchlight will chain all the results from the different search methods that we have, if a method was called before `search_author` and we don't reuse query, all the previous conditions added to the query will be lost. 

What about adding `unscope(:published_at)` here so we remove that condition?
```suggestion
          query.where(decidim_author_id: options[:current_user], decidim_author_type: Decidim::UserBaseEntity.name).unscope(:published_at)
```

Reply:
That's it @oriolgual

Reply:
Awesome! I didn't know there's a way to unscope. TY both! I'll fix this right now 😃